### PR TITLE
fix(react-query): return same `Proxy`-object to avoid infinite recursion

### LINKED ISSUE #5808: bug: Infinite render loop when referencing utils as dependency
### Provide environment information

  System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M2 Pro
    Memory: 91.06 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.19.0 - ~/Library/Caches/fnm_multishells/3059_1718376229215/bin/node
    npm: 10.2.3 - ~/Library/Caches/fnm_multishells/3059_1718376229215/bin/npm
  Browsers:
    Brave Browser: 121.1.62.165
    Chrome: 126.0.6478.62
    Edge: 126.0.2592.61
    Safari: 17.5
  npmPackages:
    @tanstack/react-query: ^5.39.0 => 5.45.1
    @trpc/client: next => 11.0.0-rc.413+87f0cbc94
    @trpc/next: next => 11.0.0-rc.413+87f0cbc94
    @trpc/react-query: next => 11.0.0-rc.413+87f0cbc94
    @trpc/server: next => 11.0.0-rc.413+87f0cbc94
    next: ^14.2.1 => 14.2.4
    react: ^18.3.0 => 18.3.1
    typescript: ^5.4.2 => 5.4.5

### Describe the bug

When hook dependencies contain the result of the `trpc.useUtils` call, react will run into an infinite loop.

Expectation:

The returned objects from trpc.useUtils are stable and can be used as dependencies.

### Link to reproduction

https://github.com/Csszabi98/trpc-infinite-loop-bug

### To reproduce

```typescript
const hello = api.post.hello.useQuery({ text: "from tRPC" });
const utils = trpc.useUtils()

useEffect(() => {
   utils.post.hello.invalidate()
}, [utils.post.hello])

```

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

## Funding
* You can sponsor this specific effort via a [Polar.sh](https://polar.sh) pledge below
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/trpc/trpc/issues/5808">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/trpc/trpc/issues/5808/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by KATT**
Closes #5808

## 🎯 Changes

This is a bit of a mindfuck.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/react-query/src/shared/proxy/utilsProxy.ts [hunk 1]
```diff
@@ -355,16 +356,27 @@ export function createReactQueryUtils<TRouter extends AnyRouter, TSSRContext>(
 ) {
   type CreateReactUtilsReturnType = CreateReactUtils<TRouter, TSSRContext>;
 
+  const clientProxy = createTRPCClientProxy(context.client);
```
Comment (KATT) [inline packages/react-query/src/shared/proxy/utilsProxy.ts:359]: creates the proxy outside of the callback function so it's not recreated on every property access

File: packages/tests/server/react/regression/issue-5808-proxy-in-dep-array.test.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { render, waitFor } from '@testing-library/react';
+import { initTRPC } from '@trpc/server';
+import { konn } from 'konn';
+import * as React from 'react';
+
+/**
+ * For reference,
+ * @link https://github.com/trpc/trpc/issues/4519
+ */
+
```
Comment (KATT) [inline packages/tests/server/react/regression/issue-5808-proxy-in-dep-array.test.tsx:11]:
```suggestion
```

File: packages/server/src/unstable-core-do-not-import/utils.ts [hunk 3]
```diff
@@ -56,3 +56,8 @@ export function isAsyncIterable<TValue>(
     asyncIteratorsSupported && isObject(value) && Symbol.asyncIterator in value
   );
 }
+
+/**
+ * Run an IIFE
+ */
+export const run = <TValue>(fn: () => TValue): TValue => fn();
```
Comment (KATT) [inline packages/server/src/unstable-core-do-not-import/utils.ts:63]:
```suggestion
```

**Review by Nick-Lucas — APPROVED** [review]
LGTM!

**Pull request merged**